util/miner_utils.go

```diff
@@ -0,0 +1,107 @@
+package util
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"net"
+	"net/netip"
+	"sync"
+	"time"
+
+	"github.com/INFURA/go-ethlibs/jsonrpc"
+
+	"github.com/dominant-strategies/go-quai-stratum/rpc"
+	"github.com/dominant-strategies/go-quai/core/types"
+)
+
+type MinerSession struct {
+	proto string
+	ip    netip.Addr
+	port  int
+	conn  *net.TCPConn
+	enc   *json.Encoder
+
+	// Stratum
+	sync.Mutex
+	latestId uint64
+}
+
+const (
+	c_Max_Req_Size = 4096
+)
+
+func NewMinerConn(endpoint string) (*MinerSession, error) {
+	localaddr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+	}
+
+	remoteaddr, err := net.ResolveTCPAddr("tcp", endpoint)
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+		panic(err)
+	}
+
+	server, err := net.DialTCP("tcp", localaddr, remoteaddr)
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+		panic(err)
+	}
+	server.SetDeadline(time.Time{})
```

Comment:
setting deadline with empty time input?

Reply:
This sets the connection to never timeout, but is default so no need to keep this line.
Removed.